src/or/hs_service.c

```diff
@@ -1638,6 +1711,20 @@ build_service_desc_keys(const hs_service_t *service,
     ret = -1;
   }
 
+  /* No need for extra strong, this is a temporary key only for this
+   * descriptor. Nothing long term. */
+  if (curve25519_keypair_generate(&desc->auth_ephemeral_kp, 0) < 0) {
```

Comment:
Do we need to create that key if we have _no_ client authorization?

Reply:
The spec says that
```
     "desc-auth-ephemeral-key" SP key NL

      [Exactly once]

      This field contains an ephemeral x25519 public key generated by the
      hidden service and encoded in base64. The key is used by the encryption
      scheme below.

      If client authorization is disabled, the value here should be a fresh
      x25519 pubkey that will remain unused.
```